netbox/project-static/src/dashboard.ts

```diff
@@ -4,15 +4,36 @@ import { apiPatch, hasError } from './util';
 
 function lockDashboard(): void {
   const dashboard = document.getElementById('dashboard') as any;
+  const gridUnlockButton = document.getElementById('unlock_dashboard') as HTMLButtonElement;
+  const gridLockButton = document.getElementById('lock_dashboard') as HTMLButtonElement;
+
   if (dashboard) {
     dashboard.gridstack.disable();
+    // Hide the grid stack items buttons
+    Array.from(document.getElementsByClassName("grid-stack-item-button")).forEach(function (element) {
+      element.classList.add('invisible');
+    });
+  }
+  if (gridUnlockButton && gridLockButton) {
+    gridUnlockButton.classList.remove('invisible');
+    gridLockButton.classList.add('invisible');
   }
 }
 
 function unlockDashboard(): void {
   const dashboard = document.getElementById('dashboard') as any;
+  const gridUnlockButton = document.getElementById('unlock_dashboard') as HTMLButtonElement;
+  const gridLockButton = document.getElementById('lock_dashboard') as HTMLButtonElement;
   if (dashboard) {
     dashboard.gridstack.enable();
+    // Show the grid stack items buttons
+    Array.from(document.getElementsByClassName("grid-stack-item-button")).forEach(function (element) {
+      element.classList.remove('invisible');
+    });
+  }
+  if (gridUnlockButton && gridLockButton) {
+    gridUnlockButton.classList.add('invisible');
+    gridLockButton.classList.remove('invisible');
   }
 }
```

Comment:
```suggestion
function lockDashboard(): void {
  const dashboard = document.getElementById('dashboard') as any;
  if (dashboard) {
    dashboard.gridstack.disable();
    // Hide the grid stack items buttons
    document.querySelectorAll(".grid-stack-item-button, #unlock_dashboard, #lock_dashboard").forEach(el => el.classList.toggle('invisible'));
  }
}

function unlockDashboard(): void {
  const dashboard = document.getElementById('dashboard') as any;
  if (dashboard) {
    dashboard.gridstack.enable();
    // Show the grid stack items buttons
    document.querySelectorAll(".grid-stack-item-button, #unlock_dashboard, #lock_dashboard").forEach(el => el.classList.toggle('invisible'));
  }
}
```

Just toggling the class seems simpler and works on my end, but not sure if I'm missing any edge cases.

Reply:
I really like your approach, I'm not seeing any obvious edge cases other than the browser potentially loosing track of the states (shouldn't happen) and making both the lock and unlock buttons visible? 

In any case I'm going to commit your approach to this PR